### PR TITLE
distinguishing javadoc comments from multiline comments

### PR DESCRIPTION
**Pull request by delgrecoj**
This pull request addresses issue #54, distinguishing javadoc comments from regular multiline comments.  The final solution ended up being a negative lookahead trick in the regular expression, and all testcases (relating to the comments module, anyway...) are passing.  Tried numerous alternatives that failed in various ways before arriving at the regular expressions used in the latest version.

Also fixed various flake8 issues, and merged with master, so this *should* merge without conflict.  Some tests are still failing, but those that are have nothing to do with the comment related code, but rather relate to other code that shouldn't have made it through to master anyway.

### CODE REVIEW

**Comment by delgrecoj** [comment]
(@yeeunmariakim, I'm intending to also implement the function for issue #56 here as well, so please wait until that has been finished to merge this.)

(seems worthwhile to keep all the refactoring of the comment functions in one place / one merge)

**Review by mariakimheinert — APPROVED** [review]

**Review by burrowss — APPROVED** [review]

**Review by bonnettd — APPROVED** [review]

**Review by karolvargas — APPROVED** [review]

**Pull request merged**
